tools/workload/benchmark_velox/native_sql_initialize.ipynb

```diff
@@ -788,27 +818,23 @@
     "        self.query_ids = sorted(self.query_infos.keys(), key=lambda x: str(len(x))+x if x[-1] != 'a' and x[-1] != 'b' else str(len(x)-1) + x)\n",
     "        print(\"http://{}:18080/history/{}/jobs/\".format(local_ip, self.sc.applicationId))\n",
     "    \n",
-    "    def start_monitor(self, clients, **kw):\n",
-    "        startmonitor(clients, self.appid, **kw)\n",
+    "    def start_monitor(self, clients, emon_list='', **kw):\n",
+    "        if emon_list:\n",
+    "            self.collect_emon = True\n",
+    "        startmonitor(clients, self.appid, self.collect_emon, **kw)\n",
     "    \n",
     "    def stop_monitor(self, clients, **kw):\n",
     "        if self.stopped:\n",
     "            return\n",
-    "        stopmonitor(clients, self.sc, self.appid, **kw)\n",
+    "        stopmonitor(clients, self.sc, self.appid, self.collect_emon, **kw)\n",
     "        if self.server:\n",
     "            output_nb = f'{self.nb_name[:-6]}-{self.appid}.ipynb'\n",
-    "            if output_nb.startswith(home):\n",
-    "                output_nb_name = os.path.relpath(output_nb, home)\n",
-    "            else:\n",
-    "                output_nb_name = output_nb\n",
-    "            output_nb_dir = os.path.dirname(output_nb_name)\n",
-    "            server_nb_dir = os.path.join('PAUS', self.base_dir, output_nb_dir)\n",
-    "            !ssh {self.server} \"mkdir -p {server_nb_dir}\"\n",
-    "            !scp {output_nb} {self.server}:{server_nb_dir}\n",
-    "            self.finished_nb = f\"http://{self.server}:8888/tree/{self.base_dir}/{output_nb_name}\"\n",
+    "            if output_nb.startswith(cwd):\n",
+    "                output_nb = os.path.relpath(output_nb, home)\n",
+    "                self.finished_nb = f\"http://{localhost}:8888/tree/{output_nb}\"\n",
```

Comment:
Just make sure this is not a typo.
I can find there are two jupyter notebook will be launched(8888 and 8889) in the initialize.ipynb.
8888 for user usage, 8889 for perf. analysis result, right?

Reply:
Yes, port 8888 is used for accessing benchmark scripts and 8889 for perf analysis scripts. We need to start the jupyter server for perf analysis script under virtual environment, as described in initialized.ipynb, therefore they cannot share the same port.